open-api/rest-catalog-open-api.yaml

```diff
@@ -3324,6 +3348,217 @@ components:
+        withoutTimezone: "2007-12-03T10:15:30"
+        withTimezone: "2007-12-03T10:15:30+00:00"
+        withFractionalSeconds: "2007-03-25T12:34:56.123456"
+
+    FixedTypeValue:
+      type: string
+      description:
+        "Fixed length type values are stored and serialized as a hexadecimal string preserving the fixed length"
+      example: "000102ff"
+
+    BinaryTypeValue:
+      type: string
+      description:
+        "Binary type values are stored and serialized as a hexadecimal string"
+      example: "000102ff"
+
+    MapTypeValue:
+      type: object
+      description:
+        "A map structure serialized with keys and values arrays that maintain type"
+      properties:
+        keys:
+          type: array
+          items:
+            $ref: '#/components/schemas/TypeValue'
+        values:
+          type: array
+          items:
+            $ref: '#/components/schemas/TypeValue'
+      example:
+        {
+          keys: [ 1, 2 ],
+          values: [ "foo", "bar" ]
+        }
+
+    StructTypeValue:
+      type: object
+      description:
+        "Struct type are serialized, where field id are preserved as a string JSON key, and the fields value is 
+        serialized based on the defined type, supporting a deep serialization of nested structures"
+      additionalProperties:
+        oneOf:
+          - $ref: '#/components/schemas/TypeValue'
+      example:
+        {
+          "4": 1,
+          "5": "bar"
+        }
+
+    ListTypeValue:
+      type: array
+      description:
+        "A list of elements, where each element is serialized according to its specific type logic"
+      items:
+         $ref: '#/components/schemas/TypeValue'
+      example: [1, 2, 3]
+
+    PrimitiveTypeValue:
+      oneOf:
+        - $ref: '#/components/schemas/BooleanTypeValue'
+        - $ref: '#/components/schemas/IntegerTypeValue'
+        - $ref: '#/components/schemas/LongTypeValue'
+        - $ref: '#/components/schemas/FloatTypeValue'
+        - $ref: '#/components/schemas/DoubleTypeValue'
+        - $ref: '#/components/schemas/DecimalTypeValue'
+        - $ref: '#/components/schemas/StringTypeValue'
+        - $ref: '#/components/schemas/UUIDTypeValue'
+        - $ref: '#/components/schemas/DateTypeValue'
+        - $ref: '#/components/schemas/TimeTypeValue'
+        - $ref: '#/components/schemas/TimestampTypeValue'
+        - $ref: '#/components/schemas/FixedTypeValue'
+        - $ref: '#/components/schemas/BinaryTypeValue'
+
+    TypeValue:
+      oneOf:
+        - $ref: '#/components/schemas/PrimitiveTypeValue'
+        - $ref: '#/components/schemas/MapTypeValue'
+        - $ref: '#/components/schemas/StructTypeValue'
+        - $ref: '#/components/schemas/ListTypeValue'
+
+    FileContent:
+      type: string
+      enum:
+        - data
+        - position-deletes
+        - equality-deletes
+
+    FileFormat:
+      type: string
+      enum:
+        - avro
+        - orc
+        - parquet
+
+    ContentFile:
```

Comment:
I think it would be better to have `DataFile` and `DeleteFile` subtypes based on this, the same way that we have `BaseUpdate` and subtypes.

That way we would not need to have `equality-field-ids` for data files. We may also want to omit parts of `ContentFile` for the `DeleteFile` types. For example, delete files have value counts and lower/upper bounds. But I'm skeptical it would ever be worth sending them. The stats for delete files are used to match them to data files and aren't needed on the client side.

And for this PR, you'd only need `DataFile`.

Reply:
I think we still need delete files? Because each file scan task will need to return 1 data file + associated delete files.

Reply:
We need delete files for the planning endpoint. We don't need them for the append endpoint.